test/integration-tests/Seaport.test.ts

```diff
@@ -141,4 +157,187 @@ describe('Seaport', () => {
       router['execute(bytes,bytes[],uint256)'](commands, inputs, DEADLINE, { value: seaportValue })
     ).to.be.revertedWith('ExecutionFailed(0, "0x8baa579f")')
   })
+
+  describe('Seaport SELL 721', async () => {
+    let tubbyCats: ERC721
+    let weth: ERC20
+
+    beforeEach(async () => {
+      await resetFork(16385143) // txn is block 16385144
+      await hre.network.provider.request({
+        method: 'hardhat_impersonateAccount',
+        params: [ALICE_ADDRESS],
+      })
+      alice = await ethers.getSigner(ALICE_ADDRESS)
+      permit2 = (await deployPermit2()).connect(alice) as Permit2
+      router = (await deployUniversalRouter(permit2)).connect(alice) as UniversalRouter
+      planner = new RoutePlanner()
+      tubbyCats = new ethers.Contract(TUBBY_ADDRESS, ERC721_ABI) as ERC721
+      weth = new ethers.Contract(WETH.address, ERC20_ABI) as ERC20
+
+      // send 1 eth from alice to router.address for weth approval
+      await alice.sendTransaction({ to: router.address, value: ethers.utils.parseEther('1.0') })
+      expect(await ethers.provider.getBalance(router.address)).to.eq(ethers.utils.parseEther('1.0'))
+      // max approve conduit for weth
+      const routerSigner = await ethers.getImpersonatedSigner(router.address)
+      await weth.connect(routerSigner).approve(OPENSEA_CONDUIT, ethers.constants.MaxUint256)
+      await weth.connect(routerSigner).allowance(router.address, OPENSEA_CONDUIT)
+      expect(await weth.connect(routerSigner).balanceOf(router.address)).to.eq(0)
```

Comment:
You can do the connect when you create weth so save you having to do connect on all of these lines :)

i.e.
```
weth = new ethers.Contract(WETH.address, ERC20_ABI).connect(routerSigner) as ERC20
```